### PR TITLE
trivial: fix various pyflakes/vulture warnings

### PR DESCRIPTION
**Pull request by eklitzke**
I am trying to learn my way around the Bitcoin code, and I decided to try to fix some minor errors reported by pyflakes/vulture. Most of these are just getting rid of unused variables.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK. Thanks for doing these uncontroversial cleanups.

Please note that the changes in this PR are partly overlapping with #10781. Would you mind reviewing #10781?

A minor nit: The changes made to `socks5.py` are whitespace only changes. According to the coding guidelines of the projects these changes should probably be excluded.

File: test/functional/rawtransactions.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def run_test(self):
         addr2Obj = self.nodes[2].validateaddress(addr2)
 
         mSigObj = self.nodes[2].addmultisigaddress(2, [addr1Obj['pubkey'], addr2Obj['pubkey']])
-        mSigObjValid = self.nodes[2].validateaddress(mSigObj)
+        assert self.nodes[2].validateaddress(mSigObj)
```
Comment (MarcoFalke) [inline test/functional/rawtransactions.py]: IIRC, this returns a dict and you'd need to assert on the value of the "isvalid" key.

**Review by maflcko — COMMENTED** [review]
utACK 8064f7abe6e6c99369fb9bde8190e766f5fa1dc2

File: test/functional/test_framework/script.py [hunk 2]
```diff
@@ -14,8 +14,7 @@
 import sys
 bchr = chr
 bord = ord
-if sys.version > '3':
-    long = int
+if sys.version_info.major == 3:
```
Comment (jnewbery) [inline test/functional/test_framework/script.py]: functional tests always run in Python 3. I think you can remove support for 2 entirely from this module.

File: test/functional/p2p-versionbits-warning.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def __init__(self):
     def setup_network(self):
         self.alert_filename = os.path.join(self.options.tmpdir, "alert.txt")
         # Open and close to create zero-length file
-        with open(self.alert_filename, 'w', encoding='utf8') as _:
+        with open(self.alert_filename, 'w', encoding='utf8'):
```
Comment (jnewbery) [inline test/functional/p2p-versionbits-warning.py]: Perhaps replace with https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch

**Review by jnewbery — COMMENTED** [review]
Looks good. Agree that #10781 should go in first and this be rebased on top of it.

Thanks for your contribution!

**Comment by MarcoFalke** [comment]
@eklitzke Needs rebase

**Comment by MarcoFalke** [comment]
Closing for now due to inactivity. Let me know to reopen.

**Pull request closed without merging**